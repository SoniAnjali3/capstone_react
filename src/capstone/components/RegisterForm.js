import React from 'react';
import { withFormik, Field, Form } from 'formik';
import * as Yup from 'yup';
import '../../../node_modules/bootstrap/dist/css/bootstrap.min.css';
import '../../App.css';

const RegisterForm = ({ errors, touched, isSubmitting }) => (
  <div className='container'>
    <h1>Register Yourself</h1>
    <Form>
      <div className='form-group'>
        <Field
          type='text'
          name='EmailID'
          placeholder='EmailID'
          className='form-control'
        />
        {touched.EmailID && errors.EmailID && (
          <div style={{ color: 'red' }}>{errors.EmailID}</div>
        )}
      </div>
      <div>
        <Field
          type='text'
          name='Password'
          placeholder='Password'
          className='form-control'
        />
        {touched.Password && errors.Password && (
          <div style={{ color: 'red' }}>{errors.Password}</div>
        )}
      </div>
      <br></br>
      <div className='form-group'>
        <Field
          type='text'
          name='FirstName'
          placeholder='FirstName'
          className='form-control'
        />
        {touched.FirstName && errors.FirstName && (
          <div style={{ color: 'red' }}>{errors.FirstName}</div>
        )}
      </div>
      <div className='form-group'>
        <Field
          type='text'
          name='LastName'
          placeholder='LastName'
          className='form-control'
        />
        {touched.LastName && errors.LastName && (
          <div style={{ color: 'red' }}>{errors.LastName}</div>
        )}
      </div>
      <div className='form-group'>
        <Field
          type='text'
          name='Location'
          placeholder='Location'
          className='form-control'
        />
        {touched.Location && errors.Location && (
          <div style={{ color: 'red' }}>{errors.Location}</div>
        )}
      </div>
      <div className='form-group'>
        <Field
          type='text'
          name='MobileNumber'
          placeholder='MobileNumber'
          className='form-control'
        />
        {touched.MobileNumber && errors.MobileNumber && (
          <div style={{ color: 'red' }}>{errors.MobileNumber}</div>
        )}
      </div>
      <br></br>
      <button type='submit' disabled={isSubmitting} className='btn btn-primary'>
        Submit
      </button>
    </Form>
  </div>
);

const FormikRegisterForm = withFormik({
  mapPropsToValues({
    EmailID,
    Password,
    FirstName,
    LastName,
    Location,
    MobileNumber,
  }) {
    //initial value passed from App.js will be set in these parameters
    return {
      EmailID: EmailID || '',
      Password: Password || '',
      FirstName: FirstName || '',
      LastName: LastName || '',
      Location: Location || '',
      MobileNumber: MobileNumber || '',
    };
  },
  validationSchema: Yup.object().shape({
    EmailID: Yup.string()
      .email('Email is not valid')
      .required('EmailID is required'),
    Password: Yup.string().required('Password is required'),
    FirstName: Yup.string().required('Please neter First name'),
    LastName: Yup.string().required('Last Name is required'),
    Location: Yup.string().required('Please Enter your location'),
    MobileNumber: Yup.string()
      .min(10, 'number should be 10 digits long')
      .max(10, 'number should be 10 digits long')
      .required('enter contact number'),
  }),
  handleSubmit(values, { props, resetForm, setSubmitting, setErrors }) {
    //console.log(values);
    setTimeout(() => {
      resetForm();
      // alert(JSON.stringify(values));
      // console.log('values : ' + values);
      props.onRegister(values);
      setSubmitting(false);
    }, 2000);
  },
})(RegisterForm);

export default FormikRegisterForm;

// export default class RegisterForm  extends React.Component {
//     constructor(props) {
//         super(props);

//         this.onSubmit = this.onSubmit.bind(this);
//     }

//     onSubmit(event) {
//         event.preventDefault();
//         var user = {};
//         user.EmailID = this.refs.EmailID.value;
//         user.Password = this.refs.Password.value;
//         user.FirstName = this.refs.FirstName.value;
//         user.LastName = this.refs.LastName.value;
//         user.Location = this.refs.Location.value;
//         user.MobileNumber = this.refs.MobileNumber.value;
//         this.props.onRegister(user);
//     }

//     render() {
//         return(
//             <form>
//                 <h1>Register User</h1>
//                 EmailID: &nbsp;
//                 <input type="text" ref="EmailID"></input><br/><br/>
//                 Password: &nbsp;
//                 <input type="text" ref="Password"></input><br/><br/>
//                 FirstName:
//                 <input type="text" ref="FirstName"></input><br/><br/>
//                 LastName:
//                 <input type="text" ref="LastName"></input><br/><br/>
//                 Location:
//                 <input type="text" ref="Location"></input><br/><br/>
//                 MobileNumber:
//                 <input type="text" ref="MobileNumber"></input><br/><br/>
//                 <button type="submit" value="Save" onClick={this.onSubmit} />
//             </form>
//         );
//     }
// }
