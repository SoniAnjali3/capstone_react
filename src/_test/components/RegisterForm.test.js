import React from 'react';
import { mount, shallow } from 'enzyme';
import RegisterForm from '../../capstone/components/RegisterForm';

describe('Test RegisterForm using Shallow rendering', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RegisterForm />);
  });

  it('has 6 text input elements', () => {
    expect(wrapper.find('input').length).toEqual(6);
  });

  it('has a single button', () => {
    expect(wrapper.find('button').length).toEqual(1);
  });

  it('has a single h1', () => {
    expect(wrapper.find('h1').length).toEqual(1);
  });

  it('has a render h1 correctly', () => {
    expect(wrapper.find('h1').render().text()).toEqual('Register User');
  });
});

describe('Test RegisterForm using Full DOM rendering', () => {
  let mountWrapper;

  beforeEach(() => {
    mountWrapper = mount(<RegisterForm />);
  });

  afterEach(() => {
    mountWrapper.unmount();
  });

  it('has 6 text input elements', () => {
    expect(mountWrapper.find('input').length).toEqual(6);
  });

  it('has a single button', () => {
    expect(mountWrapper.find('button').length).toEqual(1);
  });
});

describe('RegisterFormSnapShot', () => {
  let wrapperFullDOM;
  beforeEach(() => {
    wrapperFullDOM = mount(<RegisterForm></RegisterForm>);
  });
  test('RegisterForm renders correctly', () => {
    expect(wrapperFullDOM).toMatchSnapshot(); //first time takes copy of snapshot and afterwards compare with old snapshot
  });
});
