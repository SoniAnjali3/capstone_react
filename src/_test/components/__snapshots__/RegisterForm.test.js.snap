// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterFormSnapShot RegisterForm renders correctly 1`] = `
<RegisterForm>
  <form>
    <h1>
      Register User
    </h1>
    EmailID:  
    <input
      type="text"
    />
    <br />
    <br />
    Password:  
    <input
      type="text"
    />
    <br />
    <br />
    FirstName:
    <input
      type="text"
    />
    <br />
    <br />
    LastName:
    <input
      type="text"
    />
    <br />
    <br />
    Location:
    <input
      type="text"
    />
    <br />
    <br />
    MobileNumber:
    <input
      type="text"
    />
    <br />
    <br />
    <button
      onClick={[Function]}
      type="submit"
      value="Save"
    />
  </form>
</RegisterForm>
`;
